third_party/wabt/src/interp/jit/jit.cc

```diff
@@ -44,6 +48,62 @@ void JITFunction::call() const {
 
   func.func_entry = func_entry_;
   func.code(context, reinterpret_cast<void*>(start));
+
+  StackAllocator* stackAllocator = new StackAllocator(0);
+  for (ValueType result_type : result_types) {
+    stackAllocator->push(LocationInfo::typeToValueInfo(result_type));
+  }
+
+  int data_index = 0;
+  std::vector<LocationInfo>& offsets = stackAllocator->values();
+#define push_result(type)                                  \
+  results.push_back(Value::Make(*(reinterpret_cast<type*>( \
+      reinterpret_cast<u8*>(data) + offsets[data_index].value))));
+
+  for (ValueType result_type : result_types) {
+    switch (result_type) {
+      case Type::I32:
+        push_result(s32);
```

Comment:
This reverses the order, is this expected?

Reply:
What does this reverse the order of?